### PR TITLE
pkg/locker: replace perror with explicit error types

### PR DESCRIPTION
**Pull request by alrs**
This PR replaces `perror` with `PreemptedLockError` and `ContextCancelledError` so that the tests can verify correct errors with type assertions instead of `strings` comparison of `err.Error()`.

### CODE REVIEW

**Comment by alrs** [comment]
@bkeroackdsc do you have any reservations about this kind of enhancement? Would you rather see the errors not exported outside the package?

**Comment by bkeroackdsc** [comment]
Thanks @alrs ! Initially it seems OK to me but I'll look more in depth tomorrow and merge if it's ok.

File: pkg/locker/preemptive_locker_test.go [hunk 1]
```diff
@@ -140,11 +139,12 @@ func TestPreemptiveLockerCancelledContext(t *testing.T) {
 
 	// should fail to acquire the lock because context is cancelled
 	_, err = ftl2.Lock(ctx2)
-	if err == nil {
-		t.Fatalf("should have returned a context cancelled error")
-	}
-	if !strings.Contains(err.Error(), "context was cancelled") {
-		t.Fatalf("unexpected error: %v", err)
+	switch err {
```
Comment (bkeroackdsc) [inline pkg/locker/preemptive_locker_test.go]: This should be an explicit type switch.

File: pkg/locker/preemptive_locker_test.go [hunk 2]
```diff
@@ -324,11 +324,12 @@ func TestPreemptiveLockerMultiPreemption(t *testing.T) {
 		// always get cancelled.
 		defer cf2()
 		_, err := ftl2.Lock(ctx2)
-		if err == nil {
-			t.Fatalf("second lock should have returned a preempted error")
-		}
-		if !strings.Contains(err.Error(), "preempted") {
-			t.Fatalf("unexpected error: %v", err)
+		switch err {
```
Comment (bkeroackdsc) [inline pkg/locker/preemptive_locker_test.go]: Same here.

**Comment by alrs** [comment]
@bkeroackdsc I'll most likely not be able to look at this until after Kubecon, are you guys down here this week?

**Comment by bkeroackdsc** [comment]
@alrs @cleanshavenalex is there. Say hi!

File: pkg/locker/preemptive_locker_test.go (hunk 2 above)
Comment (alrs) [reply pkg/locker/preemptive_locker_test.go]: Done.

**Comment by alrs** [comment]
I didn't find @cleanshavenalex, but the requested changes from @bkeroackdsc did make it in.

**Comment by alrs** [comment]
I have a goroutine-in-tests fix lined up after this PR lands.

**Pull request merged**